example/lib/payments/widget/payments_user_input_widgets.dart

```diff
@@ -0,0 +1,223 @@
+          state: _state,
+          onTermsAcceptedChanged: _onTermsAcceptedChanged,
+        ),
+      ],
+    );
+  }
+}
+
+class _FormFields extends StatelessWidget {
+  final PaymentsState _state;
+  final TextEditingController _emailController;
+  final TextEditingController _amountController;
+  final FocusNode _amountFocusNode;
+  final VoidCallback _onAmountSubmitted;
+
+  final _decimalFilteringFormatter =
+      FilteringTextInputFormatter.allow(RegExp(r'[0-9]+[,.]{0,1}[0-9]*'));
+
+  final _filterDigitsAfterZeroFormatter =
+      TextInputFormatter.withFunction((oldValue, newValue) {
+    final invalid = newValue.text.startsWith(RegExp(r'0[1-9]'));
+    return invalid ? oldValue : newValue;
+  });
+
+  _FormFields({
+    required PaymentsState state,
+    required TextEditingController emailController,
+    required TextEditingController amountController,
+    required FocusNode amountFocusNode,
+    required VoidCallback onAmountSubmitted,
+  })  : _state = state,
+        _emailController = emailController,
+        _amountController = amountController,
+        _amountFocusNode = amountFocusNode,
+        _onAmountSubmitted = onAmountSubmitted;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = AppTheme.of(context);
+    final color = theme.color;
+    final typography = theme.typography;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          // TODO: Localisation
+          'Your info',
+          style: typography.title1,
+        ),
+        const SizedBox(
+          height: 26,
+        ),
+        Text(
+          // TODO: Localisation
+          'Email',
+          style: typography.subtitle1.copyWith(color: color.secondaryText),
+        ),
+        const SizedBox(
+          height: 8,
+        ),
+        KevinFormField(
+          controller: _emailController,
+          keyboardType: TextInputType.emailAddress,
+          suffixIcon: KevinFormFieldClearIcon(
+            text: _state.email,
+            onPressed: () => _emailController.clear(),
+          ),
+          textInputAction: TextInputAction.next,
+          onFieldSubmitted: (_) => _amountFocusNode.requestFocus(),
+          errorText: _state.emailError.orNull,
+        ),
+        const SizedBox(
+          height: 24,
+        ),
+        Text(
+          // TODO: Localisation
+          'Payment amount',
+          style: typography.subtitle1.copyWith(color: color.secondaryText),
+        ),
+        const SizedBox(
+          height: 8,
+        ),
+        KevinFormField(
+          controller: _amountController,
+          textInputAction: TextInputAction.done,
+          focusNode: _amountFocusNode,
+          keyboardType: const TextInputType.numberWithOptions(decimal: true),
+          inputFormatters: [
+            _decimalFilteringFormatter,
+            _filterDigitsAfterZeroFormatter,
+          ],
+          suffixIcon: KevinFormFieldClearIcon(
```

Comment:
Can't you encapsulate suffix setup inside `KevinFormField`, so this will be automatically done for every field? In theory, you can specify `onPressed` action from `KevinFormField` as `_controller.clear()` and take text as `_controller.text`. Additionally if you want / need to hide the suffix entirely - you can provide `shouldShowSuffixIcon` bool parameter in `KevinFormField`

Reply:
I try to make all common widgets as versatile as possible. In that particular case we could encapsulate suffix icon widget, but what if next time we want to create input field, where it's not needed? For that reason I introduced `KevinFormFieldClearIcon` to make it easier to use/not use it when needed.

Reply:
Nice point indeed